src/Catel.Core/Catel.Core.Shared/Services/Interfaces/IObjectIdGenerator.cs

```diff
@@ -29,7 +32,20 @@ public interface IObjectIdGenerator<TUniqueIdentifier>
     /// </summary>
     /// <typeparam name="TObjectType">The object type</typeparam>
     /// <typeparam name="TUniqueIdentifier">The unique identifier type</typeparam>
-    public interface IObjectIdGenerator<TObjectType, TUniqueIdentifier> : IObjectIdGenerator<TUniqueIdentifier>
+    public interface IObjectIdGenerator<in TObjectType, TUniqueIdentifier> : IObjectIdGenerator<TUniqueIdentifier>
+        where TObjectType : class
     {
+        /// <summary>
+        /// Gets the unique identifier for the specified instance.
+        /// </summary>
+        /// <param name="instance">The instance</param>
+        /// <param name="reuse">Indicates whether the id will be returned from released id pool</param>
+        /// <returns></returns>
+        TUniqueIdentifier GetUniqueIdentifierForInstance(TObjectType instance, bool reuse = false);
+
+        /// <summary>
+        /// Gets and sets the instance check interval.
+        /// </summary>
+        TimeSpan? InstanceCheckInterval { get; set; }
```

Comment:
Let's remove this one and just add a method "CheckForOutdatedInstances()" (but I prefer a better name).